crates/noirc_evaluator/src/ssa/integer.rs

```diff
@@ -324,37 +324,29 @@ pub fn block_overflow(
             }
             if let Some(modified_ins) = &modify_ins {
                 ins.operator = modified_ins.operator;
-                ins.rhs = eval.get_const(trunc_size, node::ObjectType::Unsigned(32));
+                ins.rhs = igen.get_or_create_const(trunc_size, node::ObjectType::Unsigned(32));
             }
-            update_ins_parameters(eval, ins.idx, l_new, r_new, max_r_value);
+            update_ins_parameters(igen, ins.id, l_new, r_new, max_r_value);
         }
     }
-    update_value_array(eval, b_idx, &value_map);
-    let mut anchor: HashMap<node::Operation, VecDeque<arena::Index>> = HashMap::new();
+    update_value_array(igen, block_id, &value_map);
+    // let mut anchor = HashMap::new();
+
     //We run another round of CSE for the block in order to remove possible duplicated truncates, this will assign 'new_list' to the block instructions
-    optim::block_cse(eval, b_idx, &mut anchor, &mut new_list);
+    // TODO: Type error, new_list: Vec<NodeId>, but block_cse expects blocks: Vec<BlockId>
```

Comment:
Guillaume mentioned that this is fine since we do not use jumps yet

Reply:
This is a different case than was mentioned in the PR. The other type error was with unrolling for loops and I've fixed it by passing the first instruction of the block instead of the block itself. It is good evidence this approach lets us catch bugs though :)

The case from this comment was actually from a bad translation on my part where block_cse had a parameter `block_list` that I thought was a list of blocks, where it should have actually been a list of instructions. I fixed the naming issue and type but forgot to uncomment this part.